component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java

```diff
@@ -89,29 +97,55 @@
                 @Parameter(name = "password",
                         description = "password to connect to redis server",
                         type = {DataType.STRING}, optional = true,
-                        defaultValue = "will try to connect without authentication")
+                        defaultValue = "will try to connect without authentication"),
+                @Parameter(name = "cluster.mode",
+                        description = "This will decide the redis mode. if this is false, client will connect to a " +
+                                "single redis node.",
+                        type = {DataType.BOOL},
+                        defaultValue = "false"),
+                @Parameter(name = "nodes",
```

Comment:
Cannot understand the difference between nodes and host, and why they have null as defaults ?

Reply:
In the single node implementation, we already has the 'host' and 'port' implementation. In the cluster mode, we need to have host and port for for all the master nodes, so thats why i have introduced a new new params call nodes. In a single node scenario, nodes will be null. thats why I provided null as default. In clustered scenario we will throw an error is nodes are not defined.